openedx/frontend-lib-content-components
# [FAL-3375] Feat: Video selection gallery page

The SelectImageModal component has been refactored so that it can also be used on the video selection page; and all its child components. Now this component is called SelectionModal and is used both for the image selector and in this new video selection screen. The [video_handler](https://github.com/openedx/edx-platform/blob/e5f81ee36af72e426a082a7ef49f4c84efe50c5a/cms/djangoapps/contentstore/views/videos.py#L208) api it's used to fetch videos. I created a new request hook to fetch videos.

**Falcon ticket:** [FAL-3375](https://tasks.opencraft.com/browse/FAL-3375)

**Testing instructions**

- Setup your devstack with [frontend-app-course-authoring](https://github.com/openedx/frontend-app-course-authoring) with the branch of: https://github.com/openedx/frontend-app-course-authoring/pull/461
- Enable the video upload pipeline with:
     - ENABLE_VIDEO_UPLOAD_PIPELINE feature flag
     - VideoUploadsEnabledByDefault enabled
- Run `make studio-static`
- Run `make build` on `frontend-app-course-authoring`
- Login on studio with a super user.
- Go to [Django Admin](http://localhost:18010/admin/edxval/coursevideo/) and create a conection with a video and the Demo course.
    - Edit the duration of the videos.
- Go to the [video upload page](http://localhost:18010/videos/course-v1:edX+DemoX+Demo_Course) and verify that there is a video uploaded.
- Go to the new [video selection gallery page](http://localhost:2001/course/course-v1:edX+DemoX+Demo_Course/videos).
- Check the design
- Verify that the video is displayed
- You can generate the error state adding a `/` on the [end of this url](https://github.com/openedx/frontend-lib-content-components/pull/278/files#diff-f90d5ad8f30cb224dad493660d60568b96bee63e8fb33595e1af113f7374dbedR71)

**Screenshots**

<details>
<summary>See all...</summary>

![Captura de pantalla de 2023-03-20 16-56-25](https://user-images.githubusercontent.com/11827305/226474856-13c71532-50fe-46ae-8f80-78f850adbb1b.png)

</details>


**Author notes**

- Missing to implement the functionality of the "Hide selected course videos" switch. The API does not bring any data that can be used to do this filtering.
- Missing to implement the upload of videos and the selection of the video (next button)
- I used [this component](https://github.com/openedx/frontend-lib-content-components/pull/275/files#diff-bbca58b1e98ac1ee967191ea49727306efa17e545d8fcdcde45132ff2ecc8c5d) from this PR: https://github.com/openedx/frontend-lib-content-components/pull/275

**Release notes**

- [ ] Merge this before: https://github.com/openedx/frontend-app-course-authoring/pull/461
